payu/branch.py

```diff
@@ -36,10 +36,23 @@
 Where BRANCH_NAME is the name of the branch"""
 
 
-def check_restart(restart_path: Optional[Path],
-                  archive_path: Path) -> Optional[Path]:
-    """Checks for valid prior restart path. Returns resolved restart path
-    if valid, otherwise returns None"""
+def check_restart(restart_path: Path,
+                  archive_path: Optional[Path] = None) -> Optional[Path]:
```

Comment:
Yes because at the time immediately after the clone, `archive` path does not exist or is unknown, and not necessarily on the same branch payu will checkout. This function is called later in the checkout method to check whether there is pre-existing restarts in the archive. I re-used the same function as it makes sense to keep the checks together and just do an extra check if archive is passed along.